src/libraries/System.Collections.Concurrent/src/System/Collections/Concurrent/BlockingCollection.cs

```diff
@@ -1353,7 +1353,7 @@ private static int TryTakeFromAnyCore(BlockingCollection<T>[] collections, [Mayb
             for (int i = 0; i < collections.Length; i++)
             {
                 // Check if the collection is not completed, and potentially has at least one element by checking the semaphore count
-                if (!collections[i].IsCompleted && collections[i]._occupiedNodes.CurrentCount > 0 && collections[i].TryTake(out item))
+                if (!collections[i].IsCompleted && collections[i]._occupiedNodes.CurrentCount > 0 && collections[i].TryTake(out item, 0, externalCancellationToken))
```

Comment:
Ditto.

Better yet, let's fix the rule and then not change the code unnecessarily.

Reply:
Okay sure, I'll fix [the rule](https://github.com/dotnet/roslyn-analyzers/issues/3786) and avoid making these changes.